openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/BallerinaTypesGenerator.java

```diff
@@ -65,9 +66,24 @@
  */
 public class BallerinaTypesGenerator {
 
-    private List<TypeDefinitionNode> typeDefinitionNodeList;
+    private List<TypeDefinitionNode> typeDefinitionNodeList = new LinkedList<>();
```

Comment:
Can we make this final and always initialize within the contructor instead?

Reply:
Updated